tests/components/raycaster.test.js

```diff
@@ -0,0 +1,24 @@
+/* global assert, process, setup, suite, test */
+var entityFactory = require('../helpers').entityFactory;
+
+suite('raycaster', function () {
+  'use strict';
+
+  setup(function (done) {
+    var el = this.el = entityFactory();
+    var sceneEl = document.querySelector('a-scene');
+    // Mocks canvas
+    sceneEl.canvas = document.createElement('div');
+    el.setAttribute('raycaster', '');
+    if (el.hasLoaded) { done(); }
+    el.addEventListener('loaded', function () {
+      done();
+    });
+  });
+
+  suite('init', function () {
+    test('the raycaster component is initalized', function () {

```

Comment:
Yeah, don't have to prepend every test with "the raycaster component" in the context of the raycaster suite though. I like to think of it reading:

```suite('raycaster') suite('init') test('initializes raycaster')``` -> `raycaster.init initializes raycaster`